Justfile

```diff
@@ -0,0 +1,183 @@
+    --dhcp-configurations "Key=domain-name,Values={{env}}-testnet-infra.local" "Key=domain-name-servers,Values=$ips" | \
+    jq -r '.DhcpOptions.DhcpOptionsId')
+  aws ec2 associate-dhcp-options --dhcp-options-id $dhcp_options_set_id --vpc-id $vpc_id
+  echo "Done"
+
+clean-opensearch-services env:
+  #!/usr/bin/env bash
+  set -e
+
+  cd networking
+  terraform workspace select {{env}}
+  testnet_infra_public_subnet_ids=$(terraform output -json public_subnet_ids)
+  cd ..
+
+  cd opensearch-networking
+  terraform workspace select {{env}}
+  data_prepper_config_efs_id=$(terraform output -raw data_prepper_config_efs_id | xargs)
+  data_prepper_security_group_id=$(terraform output -raw data_prepper_security_group_id | xargs)
+  data_prepper_service_registry_arn=$(terraform output -raw data_prepper_service_registry_arn | xargs)
+  telemetry_collector_config_efs_id=$(terraform output -raw telemetry_collector_config_efs_id | xargs)
+  telemetry_collector_target_group_arn=$(terraform output -raw telemetry_collector_target_group_arn | xargs)
+  telemetry_collector_security_group_id=$(terraform output -raw telemetry_collector_security_group_id | xargs)
+  telemetry_collector_service_registry_arn=$(terraform output -raw telemetry_collector_service_registry_arn | xargs)
+  cd ..
+
+  just init "{{env}}" "opensearch-services"
+  cd opensearch-services
+  terraform workspace select {{env}}
+  terraform destroy -auto-approve \
+    -var data_prepper_config_efs_id=$data_prepper_config_efs_id \
+    -var data_prepper_security_group_id=$data_prepper_security_group_id \
+    -var data_prepper_service_registry_arn=$data_prepper_service_registry_arn \
+    -var telemetry_collector_config_efs_id=$telemetry_collector_config_efs_id \
+    -var telemetry_collector_target_group_arn=$telemetry_collector_target_group_arn \
+    -var telemetry_collector_security_group_id=$telemetry_collector_security_group_id \
+    -var telemetry_collector_service_registry_arn=$telemetry_collector_service_registry_arn \
+    -var "public_subnet_ids=$testnet_infra_public_subnet_ids"
+
+clean-opensearch-networking env:
+  #!/usr/bin/env bash
+  set -e
+
+  cd networking
+  terraform workspace select {{env}}
+  testnet_infra_vpc_id=$(terraform output -raw vpc_id | xargs)
+  testnet_infra_security_group_id=$(terraform output -raw testnet_infra_security_group_id | xargs)
+  testnet_infra_public_subnet_ids=$(terraform output -json public_subnet_ids)
+  cd ..
+
+  just init "{{env}}" "opensearch-networking"
+  cd opensearch-networking
+  terraform workspace select {{env}}
+  terraform destroy -auto-approve \
+    -var testnet_infra_vpc_id=$testnet_infra_vpc_id \
+    -var testnet_infra_security_group_id=$testnet_infra_security_group_id \
+    -var "testnet_infra_public_subnet_ids=$testnet_infra_public_subnet_ids"
+  cd ..
+
+clean-opensearch env:
+  #!/usr/bin/env bash
+  set -e
+
+  just clean-opensearch-services "{{env}}"
+  just clean-opensearch-networking "{{env}}"
```

Comment:
Just observed that when you call `just opensearch` and then `just clean-openserach`, the cleanup fails because it assumes that the `services, networking` are up and running as well. This is not really a bug, but I faced this issue when I wanted to restart the entire process in the middle, as I had messed something up…

Reply:
Thanks for pointing that out. I now have a `clean-opensearch` and `clean-opensearch-stack` targets so that Opensearch can be deleted on its own.